vcd/resource_vcd_vm_internal_disk_test.go

```diff
@@ -0,0 +1,479 @@
+		task, err := vm.PowerOn()
+		if err != nil {
+			t.Skip(fmt.Sprintf("Error powering up vm %s", err))
+			return
+		}
+		_ = task.WaitTaskCompletion()*/
+	internalDiskSize := 20000
+
+	storageProfile := testConfig.VCD.ProviderVdc.StorageProfile
+	/*	if *adminVdc.UsesFastProvisioning {
+		// to avoid `Cannot use multiple storage profiles in a fast-provisioned VDC` we need to reuse VM storage profile
+		storageProfile = vm.VM.StorageProfile.Name
+	}*/
+
+	diskResourceName := "disk1"
+	diskSize := "13333"
+	biggerDiskSize := "14333"
+	busType := "sata"
+	busNumber := "1"
+	unitNumber := "0"
+	allowReboot := true
+
+	vappName := "TestInternalDiskVapp"
+	vmName := "TestInternalDiskVm"
+	vdcName := "ForInternalDiskTest"
+	var params = StringMap{
+		"Org": testConfig.VCD.Org,
+		//"VDC": testConfig.VCD.Vdc,
+		//		"VappName":           vapp.VApp.Name,
+		//		"VmName":             vm.VM.Name,
+		"FuncName":           "TestVappVmDS",
+		"Tags":               "vm",
+		"DiskResourceName":   diskResourceName,
+		"Size":               diskSize,
+		"SizeBigger":         biggerDiskSize,
+		"BusType":            busType,
+		"BusNumber":          busNumber,
+		"UnitNumber":         unitNumber,
+		"StorageProfileName": storageProfile,
+		"AllowReboot":        allowReboot,
+
+		"VdcName":                   vdcName,
+		"OrgName":                   testConfig.VCD.Org,
+		"AllocationModel":           "ReservationPool",
+		"ProviderVdc":               testConfig.VCD.ProviderVdc.Name,
+		"NetworkPool":               testConfig.VCD.ProviderVdc.NetworkPool,
+		"Allocated":                 "1024",
+		"Reserved":                  "1024",
+		"Limit":                     "1024",
+		"ProviderVdcStorageProfile": testConfig.VCD.ProviderVdc.StorageProfile,
+		// cause vDC ignores empty values and use default
+		"MemoryGuaranteed": "1",
+		"CpuGuaranteed":    "1",
+
+		"Catalog":      testSuiteCatalogName,
+		"CatalogItem":  testSuiteCatalogOVAItem,
+		"VappName":     vappName,
+		"VmName":       vmName,
+		"ComputerName": vmName + "Unique",
+
+		"InternalDiskSize": internalDiskSize,
+	}
+	params["FuncName"] = t.Name() + "-IdeCreate"
+	configTextIde := templateFill(sourceTestVmInternalDiskIde, params)
+	params["FuncName"] = t.Name() + "-CreateALl"
+	configText := templateFill(sourceTestVmInternalDisk, params)
+	params["FuncName"] = t.Name() + "-Update1"
+	configText_update1 := templateFill(sourceTestVmInternalDisk_Update1, params)
+	params["FuncName"] = t.Name() + "-Update2"
+	//configText_update2 := templateFill(sourceTestVmInternalDisk_Update2, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText+configText_update1)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:      configTextIde,
+				ExpectError: regexp.MustCompile(`.*The attempted operation cannot be performed in the current state \(Powered on\).*`),
```

Comment:
- Why is this an expected error?
- Which operation do we expect to fail?
- Why we can't make it succeed?

Reply:
This test dedicated to check for `allow_vm_reboot`. IDE disk's can't be changed when VM runing.

Reply:
Please put a comment that explains it. And add a skipping tag so it won't run in the binary tests.

Reply:
done